File: in_Java/0621 Task Scheduler.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] count = new int[26];
        int maxcount = 0;
        int maxfreq = 0;
        for(char c : tasks){
            count[c-'A']++;
            if(maxcount == count[c-'A']) maxfreq++;
            if(maxcount < count[c-'A']){
                maxcount = count[c-'A'];
                maxfreq = 1;
            }
        }
        int partCount = maxcount - 1;
        int emptySlots = partCount * (n - maxfreq + 1);
        int available_tasks = tasks.length - maxfreq * maxcount;
        int idles = Math.max(0, emptySlots - available_tasks);
        return idles + tasks.length;
    }
}

File: in_Python/0621 Task Scheduler.py
class Solution(object):
    def leastInterval(self, tasks, n):
        """
        :type tasks: List[str]
        :type n: int
        :rtype: int
        """
        N = len(tasks)
        d = {}
        for i in tasks:
            if i not in d:
                d[i] = 1
            else:
                d[i] += 1
        maxcount = 0
        for k in d:
            if d[k] > maxcount:
                maxcount = d[k]
        maxtask = []
        for k in d:
            if d[k] == maxcount:
                maxtask.append(k)
        L = len(maxtask)
        le = (n+1) * (maxcount-1) + L
        return max(N,le)
        
        
        